modules/ting/ting.module

```diff
@@ -1009,27 +1009,12 @@ function _ting_default_reservable_types() {
  *   Array with data well material types.
  */
 function _ting_fetch_well_types() {
-  // Get a list of types by executing a null query and look at the facets
-  // result.
-  $options = array(
-    'facets' => array('facet.type'),
-    'numFacets' => 99,
-    'reply_only' => TRUE,
-    'sort' => 'random',
-  );
-  module_load_include('client.inc', 'ting');
-  $result = ting_do_search("*", 0, 0, $options);
-
   $types = array();
-  foreach ($result->facets['facet.type']->terms as $term => $count) {
-    $types[drupal_strtolower($term)] = $count;
-  }
 
-  // Only save if we actually got any types.
-  if (!empty($types)) {
-    variable_set('ting_well_types', $types);
+  // Delegate the fetching of types to a search provider.
+  if (ding_provider_implements('search', 'material_types')) {
+    $types = ding_provider_invoke('search', 'material_types');
```

Comment:
Since this module now uses a provider function we should add [a `hook_ding_provider_user` hook declaring that](https://github.com/ding2/ding2/blob/master/modules/ding_reservation/ding_reservation.module#L227-L237).